src/keyboard-x11.c

```diff
@@ -564,9 +564,159 @@ static void onUserChanged (GDBusConnection *pConnection, const gchar *sSender, c
     }
 }
 
+static void getSystemLayouts (gchar *sLocation, Keyboard *self)
+{
```

Comment:
Can't we avoid this code duplication? E.g. by moving getSystemsLayouts into libayatana-common???